stac_fastapi/pgstac/db.py

```diff
@@ -47,33 +46,46 @@ async def con_init(conn):
 ConnectionGetter = Callable[[Request, Literal["r", "w"]], AsyncIterator[Connection]]
 
 
+async def _create_pool(settings: PostgresSettings) -> Pool:
+    """Create a connection pool."""
+    return await asyncpg.create_pool(
+        settings.connection_string,
+        min_size=settings.db_min_conn_size,
+        max_size=settings.db_max_conn_size,
+        max_queries=settings.db_max_queries,
+        max_inactive_connection_lifetime=settings.db_max_inactive_conn_lifetime,
+        init=con_init,
+        server_settings=settings.server_settings.model_dump(),
+    )
+
+
 async def connect_to_db(
     app: FastAPI,
     get_conn: Optional[ConnectionGetter] = None,
     postgres_settings: Optional[PostgresSettings] = None,
+    add_write_connection_pool: bool = False,
+    write_postgres_settings: Optional[PostgresSettings] = None,
 ) -> None:
     """Create connection pools & connection retriever on application."""
-    app_settings = app.state.settings
-
     if not postgres_settings:
         postgres_settings = PostgresSettings()
 
-    if app_settings.testing:
-        readpool = writepool = postgres_settings.testing_connection_string
-    else:
-        readpool = postgres_settings.reader_connection_string
-        writepool = postgres_settings.writer_connection_string
+    app.state.readpool = await _create_pool(postgres_settings)
+
+    if add_write_connection_pool:
+        if not write_postgres_settings:
+            write_postgres_settings = PostgresSettings()
```

Comment:
Here we are creating a new instance of PostgresSettings if not was passed, but should we default to `postgres_settings` which could be provided by the user?

Reply:
if we're making a breaking change in postgres environment variables, let's make them all the standardized variables used by all the core pg tooling too (this has always driven me crazy) PGDATABASE, PGPOOL, PGUSER, PGPASSWORD, PGPORT, PGHOST

Reply:
😅 spend some time tracking when those previous variable names where introduced: https://github.com/stac-utils/stac-fastapi/blob/1509c24f46deeb8ae79dbc16fc24c193454967f3/stac_api/settings.py#L12-L18

@geospatial-jeff any reason why you choose `POSTGRES_PASSWORD` instead of regular `PGPASSORD`? 

FYI: we do use the same names in `tipg` and `titiler-pgstac` 😓

If I understand correctly `POSTGRES_*` variable are usually used by postgres/postgis docker image when launching the container, while `PG*` should be variable used by application connecting to the DB

Reply:
Yes, the POSTGRES_* variables are used when configuring a docker image, while the PG* are used while connecting to a database.